### PR TITLE
Remove deprecated code is AWS provider

### PR DESCRIPTION
**Pull request by vincbeck**
Min Airflow version in providers moved to 2.8.0 in #41396. Some backward compatibility code in the Amazon provider package is only if the Airflow version is 2.7.1. Therefore, we can do some cleanup.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/sensors/batch.py [hunk 1]
```diff
@@ -87,17 +87,9 @@ def poke(self, context: Context) -> bool:
...
-            raise AirflowException(message)
-
-        # TODO: remove this if block when min_airflow_version is set to higher than 2.7.1
-        message = f"Batch sensor failed. Unknown AWS Batch job status: {state}"
-        if self.soft_fail:
-            raise AirflowSkipException(message)
-        raise AirflowException(message)
+            raise AirflowException(f"Batch sensor failed. AWS Batch job status: {state}")
+
+        raise AirflowException(f"Batch sensor failed. Unknown AWS Batch job status: {state}")
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/sensors/batch.py]: Just to prove I actually reviewed it :P  Feel free to reject.


```suggestion
     raise AirflowException(f"Batch sensor failed. {"Unknown " if state != BatchClientHook.FAILURE_STATE else ""} AWS Batch job status: {state}")

File: airflow/providers/amazon/aws/sensors/glue.py [hunk 2]
```diff
@@ -223,9 +220,6 @@ def poke(self, context: Context):
                 f": {response.get('ErrorString')}"
             )
             self.log.info(job_error_message)
-            # TODO: remove this if block when min_airflow_version is set to higher than 2.7.1
-            if self.soft_fail:
-                raise AirflowSkipException(job_error_message)
             raise AirflowException(job_error_message)
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/sensors/glue.py:223]: Not for this PR, but I keep thinking python needs a  "log and raise" function.  maybe some day we can add an "also log" flag to AirflowException.....

File: tests/providers/amazon/aws/sensors/test_athena.py [hunk 3]
```diff
@@ -73,17 +73,10 @@ def test_poke_intermediate_states(self, state, mock_poll_query_status):
         mock_poll_query_status.side_effect = [state]
         assert self.sensor.poke({}) is False
```
Comment (ferruzzi) [inline tests/providers/amazon/aws/sensors/test_athena.py:75]: This is so nice :heart_eyes:

**Review by ferruzzi — APPROVED** [review]
LGTM.  Tons of small changes, but I love seeing old code dropped.  The unit in particular look so much better now!

**Review by o-nikolas — APPROVED** [review]
Tests still aren't passing but in spirit the changes look good once things are passsing

**Review by eladkal — APPROVED** [review]

**Pull request merged**